modules/markdown/markdown.tsx

```diff
@@ -1,68 +1,101 @@
 import * as React from 'react'
-import {View} from 'react-native'
+import {
+	StyleProp,
+	TextProps,
+	TextStyle,
+	View,
+	ViewProps,
+	ViewStyle,
+	Text,
+} from 'react-native'
 import glamorous from 'glamorous-native'
 import ReactMarkdown from 'react-markdown'
 
 import propTypes from 'prop-types'
-ReactMarkdown.propTypes.containerTagName = propTypes.func
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+;(ReactMarkdown as any).propTypes.containerTagName = propTypes.func
 
 import {Paragraph, Strong, Emph, BlockQuote} from './formatting'
-import {Code, HighlightedCodeBlock as CodeBlock} from './code'
+import {Code, CodeBlock} from './code'
 import {Heading} from './heading'
 import {Link} from './link'
 import {Image} from './image'
 import {List, ListItem} from './list'
 
-const Softbreak = () => ' '
-const Hardbreak = () => '\n'
+const Softbreak = (props: TextProps) => <Text {...props}> </Text>
+const Hardbreak = (props: TextProps) => <Text {...props}>{'\n'}</Text>
 const HorizontalRule = glamorous.view({
 	width: '100%',
 	height: 1,
 	backgroundColor: 'black',
 })
 
-type MarkdownBlockEnum =
-	| 'BlockQuote'
-	| 'Code'
-	| 'CodeBlock'
-	| 'Emph'
-	| 'Heading'
-	| 'Image'
-	| 'ListItem'
-	| 'Link'
-	| 'List'
-	| 'Paragraph'
-	| 'Strong'
-	| 'ThematicBreak'
-
-type StyleSheetRule = number | any | Array<StyleSheetRule>
-
 type Props = {
-	styles?: {[key: MarkdownBlockEnum]: StyleSheetRule}
+	styles?: {
+		BlockQuote?: StyleProp<TextStyle>
+		Code?: StyleProp<TextStyle>
+		CodeBlock?: StyleProp<TextStyle>
+		Emph?: StyleProp<TextStyle>
+		Heading?: StyleProp<TextStyle>
+		Image?: StyleProp<ViewStyle>
+		ListItem?: StyleProp<TextStyle>
+		Link?: StyleProp<TextStyle>
+		List?: StyleProp<ViewStyle>
+		Paragraph?: StyleProp<TextStyle>
+		Strong?: StyleProp<TextStyle>
+		ThematicBreak?: StyleProp<TextStyle>
+	}
 	source: string
 }
 
 export class Markdown extends React.PureComponent<Props> {
-	render() {
+	render(): JSX.Element {
 		const {styles = {}, source} = this.props
 		return (
 			<ReactMarkdown
-				containerTagName={View}
+				containerTagName={View as unknown as string}
```

Comment:
Great to see the issue with upgrading (previously blocked by types) is unblocked by this.

Reply:
"hack"

I'll have to try upgrading to react-markdown@8 again, and see how it goes. 

Worst case, I write a custom markdown-dom-to-React component for us to use. (I don't need a custom parser, just a custom transformation on top of the parsed output.)